### DISCUSSION TITLE
Don't ask to add a name to all anonymous functions

### DISCUSSION THREAD

**Discussion by lcswillems** [vercel/next.js#23920] [Ideas]
# Bug report

## Describe the bug

With Next.js 9.4, Fast Refresh was working fine as soon as the page's exported function was not anonymous, even if there were anonymous functions exported elsewhere.

With Next.js 10, Next.js asks me to add a name to all my anonymous functions. It should not be the case.

## To Reproduce

Follow the instructions in this repository: https://github.com/lcswillems/next-excessive-named-exports

**Comment by awareness481** [comment]
Your app won't break if you use anonymous exports but your app won't preserve state (see [Dan Abramov's tweet](https://twitter.com/dan_abramov/status/1255229029608828928)) with Fast Refresh and you're also losing some other features like checking for rules of hooks.

Outside of that, even if Next.js error logging is great an error from an anonymous function will show as anonymous on the stack trace making it harder to debug.

See also:
1, [Relevant Next.js issue](https://github.com/vercel/next.js/issues/12891)
2. [Relevant open PR](https://github.com/vercel/next.js/pull/18047)

**Comment by lcswillems** [comment]
Fast refresh is working fine for me even if I have some anonymous functions! This is what I say in the issue.

Is there a way to disable the warning of Next.js that is irrelevant?

**Comment by meglio** [comment]
Dear team, I cannot a) ask every library developer to fix their libraries to work with Next.js Fast Refresh; and b) wait for them to complete the updates.

Why can't we just switch Fast Refresh OFF with a configuration option and move on with using custom 3rd party libraries with Next.js?

It is an issue as it's simply not a viable option to get all libraries respect requirements imposed by the Fast Refresh feature of Next.js

For instance, I'd like to turn it OFF entirely for the whole project as I am using a 3rd party in-house library that is not compatible with Fast Refresh and will not be made so in the next year at least, if at all.

500+ warnings spoil my command line and make it impossible to spot other, more important messages.
